apm-agent-plugins/apm-jdbc-plugin/src/main/java/co/elastic/apm/agent/jdbc/StatementInstrumentation.java

```diff
@@ -119,15 +184,40 @@ public static Span onBeforeExecute(@Advice.This Statement statement, @Advice.Arg
 
         @VisibleForAdvice
         @Advice.OnMethodExit(suppress = Throwable.class, onThrowable = Throwable.class)
-        public static void onAfterExecute(@Advice.Enter @Nullable Span span, @Advice.Thrown Throwable t) {
+        public static void onAfterExecute(@Advice.This Statement statement,
+                                          @Advice.Enter @Nullable Span span,
+                                          @Advice.Thrown @Nullable Throwable t,
+                                          @Advice.Return @Nullable Object returnValue) {
             if (span != null) {
+                if( t == null){
+                    // for 'executeUpdate' and 'executeLargeUpdate', we have to compute the sum as Statement.getUpdateCount()
+                    // does not seem to return the sum of all elements. As we can use instanceof to check return type
+                    // we do not need to use a separate advice.
+                    long affectedCount = 0;
+                    if(returnValue instanceof int[]){
+                        int[] array = (int[]) returnValue;
+                        for (int i = 0; i < array.length; i++) {
+                            affectedCount += array[i];
+                        }
+                    } else if ( returnValue instanceof long[]){
+                        long[] array = (long[]) returnValue;
+                        for (int i = 0; i < array.length; i++) {
+                            affectedCount += array[i];
+                        }
+                    }
```

Comment:
Extract public static method, annotate with `@VisibleForAdvice` and reuse in `ExecuteWithoutQueryInstrumentation`. As only Java core types are used, there are no classloader issues.

Reply:
I don't get that- those APIs return either `int` or `long`, not arrays...
When fixing, you will need to create two separate advices.

Reply:
should be fixed now